packages/simulation/src/convert.ts

```diff
@@ -0,0 +1,647 @@
+          reachableStates.add(transition.to)
+          stack.push(transition.to)
+        }
+      }
+    }
+  }
+  // Remove any states that are not in the set of reachable states
+  nfaGraph.states = nfaGraph.states.filter((state) => reachableStates.has(state.id))
+  // Remove any transitions that are not connected to reachable states
+  nfaGraph.transitions = nfaGraph.transitions.filter((transition) => reachableStates.has(transition.from) && reachableStates.has(transition.to))
+  return nfaGraph
+}
+
+// Graph traversal algorithm starting from the initial state that removes unreachable states and transitions
+export const removeUnreachableDFAStates = (dfaGraph: DFAGraph): DFAGraph => {
+  // Keep track of all the reachable states from the initial state
+  const reachableStates = new Set<StateID>([dfaGraph.initialState])
+  // Traverse the graph starting from the initial state
+  const stack = [dfaGraph.initialState]
+  while (stack.length > 0) {
+    const currentState = stack.pop() as StateID
+    for (const transition of dfaGraph.transitions) {
+      if (transition.from === currentState) {
+        // If the transition leads to a new state, add it to the reachable states
+        if (!reachableStates.has(transition.to)) {
+          reachableStates.add(transition.to)
+          stack.push(transition.to)
+        }
+      }
+    }
+  }
+  // Remove any states that are not in the set of reachable states
+  dfaGraph.states = dfaGraph.states.filter((state) => reachableStates.has(state.id))
+  // Remove any transitions that are not connected to reachable states
+  dfaGraph.transitions = dfaGraph.transitions.filter((transition) => reachableStates.has(transition.from) && reachableStates.has(transition.to))
+  return dfaGraph
+}
+
+// This will check to ensure that the graph passed in has valid states/transitions before continuing
+export const statesAndTransitionsPresent = (nfaGraph: FSAGraphIn): boolean => {
+  if (nfaGraph.states.length === 0 || nfaGraph.transitions.length === 0) {
+    return false
+  } else {
+    return true
+  }
+}
+
+// This will check to ensure that the graph passed in has a final state before continuing
+export const finalStateIsPresent = (nfaGraphStates: FSAState[]): boolean => {
+  const numberOfNFAStates: number = nfaGraphStates.length
+  let doesFinalExist: boolean = false
+  for (let curElem = 0; curElem < numberOfNFAStates; curElem++) {
+    if (nfaGraphStates[curElem].isFinal) {
+      doesFinalExist = true
+    }
+  }
+  if (!doesFinalExist) {
+    return false
+  } else {
+    return true
+  }
+}
+
+// This will check to ensure that the graph passed in has an initial state before continuing
+export const initialStateIsPresent = (nfaGraphInitialState: StateID): boolean => {
+  if (nfaGraphInitialState == null) {
+    return false
+  } else {
+    return true
+  }
+}
+
+// This will check if the merged state is different from the current state's merged state
+function arraysEqual (arr1, arr2) {
```

Comment:
We already have the lodash library so best to use the [isEqual](https://lodash.com/docs/4.17.15#isEqual) function from that